src/main/java/nl/hsac/fitnesse/fixture/util/selenium/SeleniumHelper.java

```diff
@@ -1019,25 +1024,77 @@ public void scrollTo(WebElement element, boolean toCenter) {
     }
 
     /**
-     * Takes screenshot of current page (as .png).
+     * Type of screenshot to take
+     */
+    protected enum ScreenshotType {
+        Viewport,
+        FullPage
+    }
+
+    /**
+     * Takes screenshot of page as currently visible to the user (as .png).
      * @param baseName name for file created (without extension),
      *                 if a file already exists with the supplied name an
      *                 '_index' will be added.
      * @return absolute path of file created.
      */
     public String takeScreenshot(String baseName) {
-        String result = null;
+        return takeScreenshot(baseName, ScreenshotType.Viewport);
+    }
 
-        WebDriver d = driver();
+    /**
+     * Takes screenshot of whole of current page (as .png).
+     * @param baseName name for file created (without extension),
+     *                 if a file already exists with the supplied name an
+     *                 '_index' will be added.
+     * @return absolute path of file created.
+     */
+    public String fullPageScreenshot(String baseName) {
+        return takeScreenshot(baseName, ScreenshotType.FullPage);
+    }
 
-        if (!(d instanceof TakesScreenshot)) {
-            d = new Augmenter().augment(d);
+    /**
+     * Takes screenshot of page (as .png).
+     * @param baseName name for file created (without extension),
+     *                 if a file already exists with the supplied name an
+     *                 '_index' will be added.
+     * @param type     type of screenshot.
+     * @return absolute path of file created.
+     */
+    protected String takeScreenshot(String baseName, ScreenshotType type) {
+        String result = null;
+        byte[] png = null;
+
+        switch (type) {
```

Comment:
Can this not just be two methods instead of switch on type (or at least make each branch just call a method)